### ISSUE TITLE
Indenting a (bulleted/ordered) list item replaces all annotations in the item and the item before it

### ISSUE THREAD

**Issue by ankon** [remirror/remirror#1481]
### Summary

Indenting a (bulleted/ordered) list item replaces all annotations in the item and the item before it.

### Steps to reproduce

1. Create a list with 3 items
2. Add an annotation on each item
3. Navigate to the middle item, and press `<tab>` to indent the item (i.e. to replace it with a new list with a single item)

(See https://codesandbox.io/s/trusting-rui-wcfnn?file=/src/App.tsx)

### Expected results

All annotations "stick" to their texts

### Actual results

1. Annotations in the middle item vanish
2. Annotations in the first item also vanish

### Possible Solution

I looked into this and believe the problem with the middle item comes from the fact that the indenting actually deletes content and then adds new content, thereby rightfully making prosemirror resolve the annotation position to "it's gone".
The `indentList` implementation is using two "replace" steps, while they should use a "replace around" step so that the original text is retained and not considered "new". See https://github.com/remirror/remirror/blob/d4bf51d6bea0cdb40665f485e22fe5e22404213c/packages/remirror__extension-list/src/list-command-indent.ts#L96-L145

It seems that dedenting actually works as expected, and retains the annotations.

I do _not_ know yet why the item _before_ the indented one is affected.

### Screenshot(s)

(see codesandbox)

**Comment by ocavue** [comment]
I agree with you. `ReplaceAroundStep` should be the solution for list indentation. I've tried to use `ReplaceAroundStep` before but I switched to delete-and-add in the end, because `ReplaceAroundStep` requires some “incomplete" `slice` during the process, which is hard (but possible) to get correct. 

I'll revisit this function and see if I can refactor it with `ReplaceAroundStep`.





---

Expanding this topic, I've found that when I try to implement the moving block feature like the following video, I can't use ReplaceAroundStep and I have to use delete-and-add. This will remove all the decoration within the moving block. Would you happen to have some suggestions about this scenario?

https://user-images.githubusercontent.com/24715727/151468555-2f7a4835-3060-4cc2-a737-c205b9ce1dd8.mp4

**Comment by ankon** [comment]
@ronnyroeller and I actually had a discussion yesterday about annotations and copy/paste (moving is just a form of that, IMHO). Our conclusion was that annotations don't work for that, they are "part of the environment", onto which the text is put. Copy/paste would require the highlighting to be part of the text itself, i.e. be a mark or similar. 

In a way that also means trying to retain annotations when modifying the text is nice, but ultimately is always a "best-effort" thing: It might not work, or it might be prohibitively expensive to do.

In your case you could maybe do a replace-around and treat everything else in the document as "around", but that would destroy annotations in the rest of the document ...
